packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/Withdraw/AddBankWireForm/index.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useRef, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import axios from 'axios'
+import { destroy, formValueSelector, reduxForm } from 'redux-form'
+
+import { getDomains } from '@core/redux/walletOptions/selectors'
+import FlyoutContainer from 'components/Flyout/Container'
+import { actions } from 'data'
+import { getFiatCurrency } from 'data/components/withdraw/selectors'
+import { getUserApiToken } from 'data/modules/profile/selectors'
+import { BankDWStepType } from 'data/types'
+
+import Loading from '../EnterAmount/template.loading'
+import ConfirmData from './Steps/ConfirmData'
+import { WIRE_BANK_FORM } from './Steps/constants'
+import EnterIntermediaryBank from './Steps/EnterIntermediaryData'
+import EnterUserData from './Steps/EnterUserData'
+import Failure from './Steps/Failure'
+import { ADD_WIRE_BANK_STEPS, WireBankFormType } from './Steps/StepsTypes'
+import Success from './Steps/Success'
+
+type FORM_PAYLOAD = {
+  accountNumber: string
+  bankName: string
+  currency: string
+  intermediaryBankInfo?: {
+    bankAccountNumber: string
+    bankName: string
+    bankRoutingNumber: string
+  }
+  product?: string
+  routingNumber: string
+}
+
+const AddWireBank = () => {
+  const [step, setStep] = useState<ADD_WIRE_BANK_STEPS>('USER_INFO')
+
+  const dispatch = useDispatch()
+
+  const formValues = useSelector((state) =>
+    formValueSelector('addWireBank')(
+      state,
+      'accountNumber',
+      'bankName',
+      'hasIntermediaryBank',
+      'intermediaryAccountNumber',
+      'intermediaryBankName',
+      'intermediaryRoutingNumber',
+      'routingNumber'
+    )
+  ) as WireBankFormType
```

Comment:
```suggestion
const formValues = useSelector(
  useMemo(
    () =>
      formValueSelector('addWireBank')(
        state,
        'accountNumber',
        'bankName',
        'hasIntermediaryBank',
        'intermediaryAccountNumber',
        'intermediaryBankName',
        'intermediaryRoutingNumber',
        'routingNumber'
      ) as WireBankFormType,
    [state] // Add dependencies as needed
  )
);
```

Consider using the useMemo hook to memoize the result of the formValueSelector to avoid unnecessary re-computations.

Reply:
Nice! It sent me digging in the docs and found out that `redux` offers a `shallowEqual` function that helps with this - see https://react-redux.js.org/api/hooks#equality-comparisons-and-updates